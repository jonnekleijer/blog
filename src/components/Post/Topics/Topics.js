// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import styles from './Topics.module.scss';

type Props = {
  topics: string[],
  topicSlugs: string[]
};

const Topics = ({ topics, topicSlugs }: Props) => (
  <div className={styles['topics']}>
    <ul className={styles['topics__list']}>
      {topicSlugs && topicSlugs.map((slug, i) => (
        <li className={styles['topics__list-item']} key={topics[i]}>
          <Link to={slug} className={styles['topics__list-item-link']}>
            {topics[i]}
          </Link>
        </li>
      ))}
    </ul>
  </div>
);

export default Topics;
