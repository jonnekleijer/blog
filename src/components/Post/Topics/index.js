// @flow strict
export { default } from './Topics';
