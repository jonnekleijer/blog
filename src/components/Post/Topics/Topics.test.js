// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Topics from './Topics';

describe('Topics', () => {
  it('renders correctly', () => {
    const props = {
      topics: [
        'test_0',
        'test_1'
      ],
      topicSlugs: [
        '/test_0',
        '/test_1'
      ]
    };

    const tree = renderer.create(<Topics {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
