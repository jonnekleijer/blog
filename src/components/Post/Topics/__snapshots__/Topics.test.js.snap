// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topics renders correctly 1`] = `
<div
  className="topics"
>
  <ul
    className="topics__list"
  >
    <li
      className="topics__list-item"
    >
      <a
        className="topics__list-item-link"
        href="/topic/test-0"
      >
        test_0
      </a>
    </li>
    <li
      className="topics__list-item"
    >
      <a
        className="topics__list-item-link"
        href="/topic/test-1"
      >
        test_1
      </a>
    </li>
  </ul>
</div>
`;
